### PR TITLE
ci: add unit testing for Swift

### PR DESCRIPTION
**Pull request by rebello95**
Add support and examples for running Swift unit tests on CI to match up with Kotlin support that was added in https://github.com/lyft/envoy-mobile/pull/231.

This also moves the Swift source code into a `src` directory to match up with Kotlin.

Resolves https://github.com/lyft/envoy-mobile/issues/248

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -284,6 +284,18 @@ stages:
...
+        pool:
+          vmImage: 'macos-10.14'
+        steps:
+          - checkout: self
+            submodules: true
+          - script: ./ci/mac_ci_setup.sh
+            displayName: 'Install dependencies'
+          - script: |
+              ./bazelw test --test_output=all --build_tests_only //library/swift/test/...
+            displayName: 'Run Swift library tests'
```
Comment (keith) [inline azure-pipelines.yml]: looks like this isn't interpreted correctly because of the multiline string above

Comment (rebello95) [reply azure-pipelines.yml]: Seems to have been working for Android, but updated

Comment (keith) [reply azure-pipelines.yml]: maybe it's just github's rendering then, but i think this is nicer

Comment (rebello95) [reply azure-pipelines.yml]: Yep same

File: azure-pipelines.yml [hunk 2]
```diff
@@ -168,9 +168,8 @@ stages:
               export CC=clang
               export CXX=clang++
             displayName: 'Set up clang toolchain'
-          - script: |
-              ./bazelw test --test_output=all --build_tests_only //library/kotlin/test/...
-            displayName: '//library/kotlin tests'
+          - script: ./bazelw test --test_output=all --build_tests_only //library/kotlin/test/...
```
Comment (buildbreaker) [inline azure-pipelines.yml]: Thanks

**Pull request merged**